protocols/v2/noise-sv2/src/cipher_state.rs

```diff
@@ -1,19 +1,91 @@
+// # Cipher State Management
+//
+// Defines the [`CipherState`] trait and the [`GenericCipher`] enum, which manage the state of
+// AEAD ciphers used in the Noise protocol. This includes managing the encryption key, nonce, and
+// the cipher instance itself, facilitating secure encryption and decryption during communication.
+//
+// The [`CipherState`] trait abstracts the management of core elements for AEAD ciphers:
+// - Manages the encryption key lifecycle used by the AEAD cipher.
+// - Generates and tracks unique nonces for each encryption operation, preventing replay attacks.
+// - Initializes the appropriate cipher (e.g., [`ChaCha20Poly1305`] or [`Aes256Gcm`]) for secure
+//   communication.
+//
+// The trait provides methods for encrypting and decrypting data using additional associated data
+// (AAD) and securely erasing sensitive cryptographic material when no longer needed.
+//
+// The [`GenericCipher`] enum enables flexible use of either [`ChaCha20Poly1305`] or [`Aes256Gcm`]
+// ciphers. It abstracts away the specific cipher being used while ensuring consistent handling of
+// cryptographic operations (e.g., encryption, decryption, key erasure) across both ciphers.
+//
+// ## Usage
+//
+// The [`CipherState`] trait is used by the [`crate::handshake::HandshakeOp`] trait to manage
+// stateful encryption and decryption tasks during the Noise protocol handshake. By implementing
+// [`CipherState`], the handshake process securely manages cryptographic material and transforms
+// messages exchanged between the initiator and responder.
+//
+// Once the Noise handshake is complete, the [`crate::Initiator`] and [`crate::Responder`] use
+// [`GenericCipher`] instances (`c1` and `c2`) to perform symmetric encryption and decryption.
+// These ciphers, initialized and managed through the [`CipherState`] trait, ensure ongoing
+// communication remains confidential and authenticated.
+//
+// The [`CipherState`] trait and [`GenericCipher`] enum are essential for managing AEAD ciphers
+// within the Noise protocol, ensuring secure data handling, key management, and nonce tracking
+// throughout the communication session.
+
 use std::ptr;
 
 use crate::aed_cipher::AeadCipher;
 use aes_gcm::Aes256Gcm;
 use chacha20poly1305::{aead::Buffer, ChaCha20Poly1305};
 
+/// The `CipherState` trait manages AEAD ciphers for secure communication, handling the encryption key, nonce,
```

Comment:
I don't think `CypherState` is exported to the outside world

Reply:
Interesting shouldn't be here. Thanks for pointing this out...